core/capabilities/triggers/logevent/trigger.go

```diff
@@ -0,0 +1,196 @@
+package logevent
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"strconv"
+	"time"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/capabilities"
+	"github.com/smartcontractkit/chainlink-common/pkg/logger"
+	"github.com/smartcontractkit/chainlink-common/pkg/types"
+	"github.com/smartcontractkit/chainlink-common/pkg/types/core"
+	"github.com/smartcontractkit/chainlink-common/pkg/types/query"
+	"github.com/smartcontractkit/chainlink-common/pkg/types/query/primitives"
+	"github.com/smartcontractkit/chainlink-common/pkg/values"
+)
+
+// Log Event Trigger Capability Request Config Details
+type RequestConfig struct {
+	ContractName      string `json:"contractName"`
+	ContractAddress   string `json:"contractAddress"`
+	ContractEventName string `json:"contractEventName"`
+	// Log Event Trigger capability takes in a []byte as ContractReaderConfig
+	// to not depend on evm ChainReaderConfig type and be chain agnostic
+	ContractReaderConfig map[string]any `json:"contractReaderConfig"`
+}
+
+// LogEventTrigger struct to listen for Contract events using ContractReader gRPC client
+// in a loop with a periodic delay of pollPeriod milliseconds, which is specified in
+// the job spec
+type logEventTrigger struct {
+	ch   chan<- capabilities.TriggerResponse
+	lggr logger.Logger
+
+	// Contract address and Event Signature to monitor for
+	reqConfig      *RequestConfig
+	contractReader types.ContractReader
+	relayer        core.Relayer
+	startBlockNum  uint64
+
+	// Log Event Trigger config with pollPeriod and lookbackBlocks
+	logEventConfig Config
+	ticker         *time.Ticker
+	cancel         context.CancelFunc
+}
+
+// Construct for logEventTrigger struct
+func newLogEventTrigger(ctx context.Context,
+	cancel context.CancelFunc,
```

Comment:
What is the purpose of passing `cancel context.CancelFunc` from the caller? It looks like all of the usages of the `ctx` that it corresponds to will be complete by the end of this func. Let's defer closing it normal on return instead.

Reply:
> It looks like all of the usages of the ctx that it corresponds to will be complete by the end of this func
No, cancel is also called in the Stop function, which is called from UnregisterTrigger, at which point we do not have access to the trigger's cancel function in the caller.

Reply:
The stop function cannot be called before this func returns, at which point cancelling has no effect.

Reply:
This function is called in a separate in a separate go routine at the callsite

Reply:
Implemented Start/Close pattern in logEventTrigger and used stopChan to track listener
Closing this based on our discussion